transport/internet/splithttp/connection.go

```diff
@@ -1,11 +1,112 @@
 package splithttp
 
 import (
+	"bytes"
+	"errors"
+	"fmt"
 	"io"
 	"net"
+	"strconv"
+	"strings"
 	"time"
 )
 
+var (
+	ErrBadRespCode = errors.New("bad response code")
+	BadCodes       = map[int]struct{}{502: {}, 503: {}, 505: {}}
+)
+
+// Optimised to read only response codes
+// Reads response codes until getting EOF or error
+func ConnHttpReadRespCodes(conn net.Conn) (codes []int, err error) {
+	buff := &bytes.Buffer{}
+	if _, err = io.Copy(buff, conn); err != nil {
+		return nil, err
+	}
+
+	for {
+		var line string
+		line, err = buff.ReadString('\n')
+		if err != nil && err == io.EOF {
+			break
+		} else if err != nil {
+			return codes, err
+		}
+
+		if strings.HasPrefix(line, "HTTP/") {
+			parts := strings.Split(line, " ")
+			if len(parts) < 2 {
+				continue
+			}
+			if code, err := strconv.Atoi(parts[1]); err == nil {
+				codes = append(codes, code)
+			}
+		}
+
+		for {
+			line, err := buff.ReadString('\n')
+			if err != nil && err == io.EOF {
+				break
+			} else if err != nil {
+				return codes, err
+			}
+			if line == "\r\n" || line == "\n" {
+				break // End of headers
+			}
+		}
+	}
+
+	return codes, err
+}
+
+// ConnHolder implements the net.Conn interface
+// adds logic of reading the responses before writing the next request
+// Used as a bugfix for HTTP1.1
+type ConnHolder struct {
+	ResponsesToRead int
+	Conn            net.Conn
```

Comment:
```suggestion
	net.Conn
```

if you do this then you don't need to forward all the methods